contracts/src/v0.8/KeeperRegistry.sol

```diff
@@ -625,7 +643,16 @@ contract KeeperRegistry is
   function getMaxPaymentForGas(uint256 gasLimit) public view returns (uint96 maxPayment) {
     (uint256 gasWei, uint256 linkEth) = _getFeedData();
     uint256 adjustedGasWei = _adjustGasPrice(gasWei, false);
-    return _calculatePaymentAmount(gasLimit, adjustedGasWei, linkEth);
+    uint256 l1CostWei = 0;
+    bytes memory data = new bytes(0);
+    if (PAYMENT_MODEL == PaymentModel.OPTIMISM) {
+      data = bytes.concat(ESTIMATED_MSG_DATA, L1_FEE_DATA_PADDING);
+      l1CostWei = OPTIMISM_ORACLE.getL1Fee(data);
+    } else if (PAYMENT_MODEL == PaymentModel.ARBITRUM) {
+      (, , , , , uint256 l1GasWei) = ARBITRUM_ORACLE.getPricesInWei();
+      l1CostWei = gasLimit * l1GasWei;
```

Comment:
interesting idea! do we know how good of an estimate this is?

Alternatively another idea i was thinking is we could call `getCurrentTxL1GasFees` and multiply it by 2 or something similar